src/extension/app.ts

```diff
@@ -311,20 +314,57 @@ class App {
         });
 
         global.display.connect('grab-op-begin', (_display: Display, win: Window) => {
-            if(validWindow(win)) {
+            const useModifier = getBoolSetting(SETTINGS.USE_MODIFIER);
+            this.isGrabbing = true;
+
+            if(validWindow(win) && !useModifier) {
                 activeMonitors().forEach(m => {
                     this.tabManager[m.index]?.show();
                 });
             }
-
         });
 
         global.display.connect('grab-op-end', (_display: Display, win: Window) => {
-            if(validWindow(win)) {
-                activeMonitors().forEach(m => {
-                    this.tabManager[m.index]?.hide();
+            const useModifier = getBoolSetting(SETTINGS.USE_MODIFIER);
+            this.isGrabbing = false;
+
+            if(!validWindow(win)) {
+                return;
+            }
+
+            activeMonitors().forEach(m => {
+                this.tabManager[m.index]?.hide();
+
+                if (!useModifier || this.modifiersManager.isHolding(MODIFIERS_ENUM.CONTROL)) {
+                    if (!trackedWindows.includes(win)) {
+                        trackedWindows.push(win);
+                    }
+
                     this.tabManager[m.index]?.moveWindowToWidgetAtCursor(win);
                     this.tabManager[m.index]?.layoutWindows();
+                    return;
+                }
+
+                if (useModifier && !this.modifiersManager.isHolding(MODIFIERS_ENUM.CONTROL) && trackedWindows.includes(win)) {
+                    trackedWindows.splice(trackedWindows.indexOf(win), 1);
+                }
+            });
+        });
+
+        this.modifiersManager.connect("changed", () => {
+            const useModifier = getBoolSetting(SETTINGS.USE_MODIFIER);
```

Comment:
Instead of checking if `useModifier == true` inside the callback, you could skip the connect method as whole. Something like:

```typescript
if (getBoolSetting(SETTINGS.USE_MODIFIER)) {
    this.modifiersManager.connect("changed", () => {
        if (!this.isGrabbing) {
            return;
        }
        // ...
```

If users change any setting, the whole extension is restarted so this has the advantage of keeping the code cleaner without drawbacks.